packages/linear/src/components/PRPopover.tsx

```diff
@@ -15,128 +14,100 @@ interface PRPopoverProps {
   className?: string
 }
 
-const PR_REFERENCE_REPLACE_KEY = '<!-- __PR_REFERENCE__ -->'
-
-const usePRsFromDescription = (issue: Issue) => {
-  const texts = issue?.description?.match(/\[.+]\(https:\/\/github\.com\/.+\/pull\/\d+.*\)/gim)
-
-  return useMemo(() => {
-    const map = new Map<string, { number: string; texts: Set<string> }>()
-    texts?.forEach((text) => {
-      const [_t, _p, url, number] = text.match(/(\()(https:\/\/github\.com\/.+\/pull\/(\d+))/i) || []
-      if (url) {
-        const item = map.get(url) || { number, texts: new Set<string>() }
-        item.texts.add(text)
-        map.set(url, item)
-      }
-    })
-
-    return map
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [texts?.toString()])
+const isPullRequest = (attachement: Attachment) =>
+  attachement.source?.type === 'github' && !!attachement.source?.pullRequestId
+
+// Returns the map of PRs in from of (number => pr metadata)
+const usePRs = (issue: Issue): Record<string, Attachment> => {
+  const fetchIssueAttachements = useQuery({
+    enabled: !!issue,
+    queryFn: async () => issue.attachments(),
+    queryKey: ['linear', 'issues', issue.identifier, 'attachements'],
+  })
+
+  const attachements = fetchIssueAttachements.data?.nodes ?? []
+
+  return Object.fromEntries(
+    attachements.filter(isPullRequest).map((attachement) => [attachement.metadata.number.toString(), attachement]),
+  )
```

Comment:
nit: same typo here 
What do you think about adding error handling and memoization:

```suggestion
const usePRs = (issue: Issue): {
  prs: Record<string, Attachment>;
  isLoading: boolean;
  error: Error | null;
} => {
  const fetchIssueAttachments = useQuery({
    enabled: !!issue,
    queryFn: async () => issue.attachments(),
    queryKey: ['linear', 'issues', issue.identifier, 'attachments'],
    onError: (error) => {
      console.error('Failed to fetch issue attachments:', error);
    },
  });

  const attachments = fetchIssueAttachments.data?.nodes ?? [];

  const prs = useMemo(
    () =>
      Object.fromEntries(
        attachments
          .filter(isPullRequest)
          .map((attachment) => [
            attachment.metadata.number.toString(),
            attachment,
          ]),
      ),
    [attachments],
  );

  return {
    prs,
    isLoading: fetchIssueAttachments.isLoading,
    error: fetchIssueAttachments.error,
  };
};
```